src/main/kotlin/Command/base/Processor.kt

```diff
@@ -1,19 +1,12 @@
 package Command.base
 
-import Command.commands.ChooseIngredientCommand
-import Command.commands.ChooseNoodlesCommand
-import Command.commands.ChooseSauceCommand
-import Command.commands.ShowPriceCommand
+class Processor(val commands: List<Command>,
+                val help: Command) {
 
-class Processor(val cart: Cart) {
-
-  fun process(command : String) : Command {
-    when (command) {
-      ChooseNoodlesCommand.CHOOSE_NOODLES -> return ChooseNoodlesCommand(cart)
-      ChooseIngredientCommand.CHOOSE_INGREDIENT -> return ChooseIngredientCommand(cart)
-      ChooseSauceCommand.CHOOSE_SAUCE -> return ChooseSauceCommand(cart)
-      else -> return ShowPriceCommand(cart)
-    }
+  fun process(command: String): Command {
+    return commands
+      .filter { it.matches(command) }
+      .getOrElse(0) { help }
```

Comment:
Well, I think I just understand why by reading again the PR.

If we add new Commands the Processor doesn't need to change.